### ISSUE TITLE
Sprint Nov 9

### ISSUE THREAD

**Issue by diasdavid** [ipfs/pm#51]
# Sprint Nov 9

## Sprint Goals

- 

## Sprint Discussions

[Schedule](https://github.com/ipfs/pm#sprint-discussion-schedule)

Please take notes in a separate pad, if you can, and link it here.

Endeavour         | Lead           | Pad    | Permanent Link
:---------------: | :------------: | :----: | :------------:
Apps on IPFS      | Juan Benet and Friedel Ziegelmayer | https://public.etherpad-mozilla.org/p/nov-9-apps-on-ipfs |  
infrastructure    | Lars Gierth    | https://public.etherpad-mozilla.org/p/nov-9-infrastructure |  
libp2p            | David Dias     | https://public.etherpad-mozilla.org/p/nov-9-libp2p |  
js-ipfs         | David Dias     | https://public.etherpad-mozilla.org/p/nov-9-js-ipfs |  
go-ipfs           | Jeromy Johnson | https://public.etherpad-mozilla.org/p/nov-9-go-ipfs | 
testing + ci      | Jeromy Johnson | https://public.etherpad-mozilla.org/p/nov-9-testing-ci |  
protocol + specs  | Juan Benet     | https://public.etherpad-mozilla.org/p/nov-9-protocol-specs |  
bitswap ml        | Juan Benet     | https://public.etherpad-mozilla.org/p/nov-9-bitswap-ml |  
Data Structures   | Juan Benet     | https://public.etherpad-mozilla.org/p/nov-9-data-structures |  

Please add the Agenda to the Pad before the endeavour sprint starts.

## Sprint Deliverables

- Add yourself below, that way we only have people listed who are really in the sprint
- Add links to issues down here. Only add things you can finish this sprint.

### [David Dias](@diasdavid)

- libp2p
  - Continue making modules work in the browser
  - [ ] Get webrtc-explorer as one of the transports _(blocked by node-webrtc not compiling on the Mac OS X, problem introduced with XCode7 and El Capitan)_
  - [ ] Sync with @whyrusleeping thoughts on how to level both implementations (on hold till go-libp2p is out of IPFS)
  - [x] Review IPLD
  - [x] make a sync API for js-ipld https://github.com/diasdavid/js-ipld/pull/8
- npm's registry on IPFS (on hold till 0.3.9 rebase on top of 0.4.0)
  - [x] Release a working version of ipfs-blob-store with all of the new shiny things https://github.com/ipfs/ipfs-blob-store/releases/tag/v0.1.0
  - [ ] download the entire npm
  - [ ] install modules via IPFS using reginabox+registry-static
- js-ipfs-api
  - [x] fix double recursive flag bug https://github.com/ipfs/js-ipfs-api/pull/110
  - [x] fix, make errors parsable again https://github.com/ipfs/js-ipfs-api/pull/111
- js-ipfs
  - [ ] Continue working on jsipfs module https://github.com/ipfs/js-ipfs/pull/35
- extra
  - [x] IPFS Lisbon
  - [x] Got invited for NodeConf Barcelona + preparing for that
  - [x] Be a NodeKnockout Judge
  - [ ] npm org for IPFS

#### IFPS HTTP docs
- [ ] Automatically populate a basic template as a one-off seed. Perhaps use ipfs/js-ipfs-api/pull/105.
- [ ] Put a Swagger info bloc in ipfs/api.
- [ ] Look at httpie
- [ ] Document how to use curl for your own use to figure out what is missing in the API docs.
- [ ] Add a PR to ipfs.io/docs/api about how the commands there are not all-inclusive.
- [ ] Get `ipfs add` documented.
- [ ] Stub out the rest.

#### READMEs
- [ ] Parse IPFS readme's automatically using a CLI tool for [ipfs-readme-standard](ipfs/ipfs-readme-standard).
- [ ] CR on @victorbjelkholm's ipfs/js-ipfs-api/pull/102
- [ ] Help with ipfs/ci-status/issues/1 and ipfs/infrastructure/issues/25.

#### Community
- [ ] Add ipfs/ci-status to PD.md
- [ ] Sprint management.
- [ ] Respond to issue about community infrastructure for meetups.
- [ ] Make the documentation around Commit Sign-Offs more friendly
- [ ] Edit ipfs/community/blob/master/docs/amending-commits.md
- [ ] Go through and move ideas from ipfs/pm/issues/43 into ipfs/pm/README.md.
- [ ] Template out issue reporting file from ipfs/go-ipfs/issues/1946.

### @lgierth

- go-ipfs
  - [~] RFCR fix gateway links with url rewrites ipfs/go-ipfs#1971 ipfs/infrastructure#116
  - [x] deploy dev0.4.0 to castor and pluto ipfs/infrastructure#115
- infrastructure
  - [x] fix restarter ipfs/infrastructure@6dfef4b
  - [x] update ansible ipfs/infrastructure#114
  - [x] remove ipfs logging config, was thought to be broken, but actually removed! ipfs/go-ipfs#1959
  - [~] open up cors on gateways (allow origin *) ipfs/infrastructure#109
  - [ ] \<hostname>.h.ipfs.io pointing to cjdns addresses
- refs
  - [ ] dmca yyyy/mm/dd structure ipfs/refs#4
  - [x] writeup for kyledrake https://asciinema.org/a/7wgjaa5co0oo1lft5l1tvws43
- cjdns
  - [x] update to latest v17
  - [~] discover peers from cjdns ipfs/go-ipfs#1316
  - [ ] invite to fc00 development (aka go-cjdns)
- org stuff
- hope
  - [ ] plan for community pinning / pincoop ipfs/infrastructure#113
  - [ ] plan for writable gateway ipfs/infrastructure#105

**Comment by diasdavid** [comment]
Endeavour leads, please update your sprint meeting pads with a proposed agenda for tomorrow

@jbenet @whyrusleeping @dignifiedquire @lgierth

**Comment by diasdavid** [comment]
Today, we will be used the newly and agreed sprint times (https://github.com/ipfs/pm/pull/50)

![image](https://cloud.githubusercontent.com/assets/1211152/11039434/93ce5e54-86ff-11e5-9f88-2145374c5035.png)

https://calendar.google.com/calendar/embed?src=ipfs.io_eal36ugu5e75s207gfjcu0ae84@group.calendar.google.com&ctz=America/New_York

**Comment by lgierth** [comment]
## Infrastructure

## Agenda

- PinCoop (formerly known as OpenIPFS)

## Participants

@jbenet
@kyledrake
@lgierth
@whyrusleeping

## Notes

- PinCoop concerns
  - we want to contribute storage
  - security concern: it requires to open up the API so that the web app can pin stuff
  - does it keep track of what gets pinned?
  - another possible solution
    - maintain a refs list of community pinnings which nodes can query and process
- 32C3
  - wiki and call for assemblies were published today
  - no concrete information on ticket sale yet, should be within the next 2 weeks though
  - should we register an assembly? https://events.ccc.de/congress/2015/wiki/Static:Assemblies
- ipfs.io/refs
  - dmca list is finally online at https://ipfs.io/refs/lists/denylists/dmca
  - nginx rewrites the url from ipfs.io/refs/ to refs.ipfs.io/ and the gateway renders broken links for that
  - more non-dmca refs stuff to come!

**Comment by Dignifiedquire** [comment]
## Apps on IPFS

#### Lead: Friedel Ziegelmayer (@dignifiedquire)

## Agenda

### Components all the things
    - https://github.com/ipfs/webui/pull/73
    - https://github.com/ipfs/ipfs/issues/109

### Time table for station release

## Participants
  - Friedel Ziegelmayer
  - David Dias
  - Johan Kiviniemi
  - Juan Benet
  
  
## Notes

- Bug on 0.3.9 passed the tests cause there wasn't a way to test a long running node (due to use of disposable deamons). We should add one test to check this (or just run tests twice without cleaning up)
- station
- ipfs-api needs to be upgraded in station - https://github.com/ipfs/station/blob/master/package.json#L10
- go-ipfs needs an option to select the OS and Arch
- what needs to be done in order to have packaging working https://github.com/ipfs/station/blob/release/gulpfile.babel.js
- option on ipfsd-ctl to select the daemon in which to control
- npm orgs for IPFS

- Moved Component chat to its own slot later this week

**Comment by diasdavid** [comment]
## libp2p

#### Lead: David Dias

## Agenda


**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

## Participants

Jeromy Johnson
David Dias
Juan benet
Johan Kiviniemi

## Notes

Jeromy is merging #1937 (Monday)
strip out libp2p out of go-ipfs (Monday)
Level up both implementations (JS and Go)
Sync up (Jeromy + David) on how to move go-libp2p to the new libp2p spec (probably happening Wednesday)
Keep working closely to node-webrtc, unblock webrtc-explorer
libp2p to be a first class citizen in the projects portfolio

**Comment by diasdavid** [comment]
## js-ipfs

#### Lead: David Dias

## Agenda


**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

## Participants

David Dias
Juan benet
Johan Kiviniemi

## Notes

- expose an HTTP API
- run sharness tests on js-ipfs HTTP API
- Review IPLD document
- ASCII(Mazières) links


##### After sprint meeting is finished, create the respective action items on the Github sprint issue

**Comment by whyrusleeping** [comment]
## @whyrusleeping TODO:
- [ ] merge transports changes
- [ ] rebase 0.4.0
- [ ] extract go-libp2p
- [ ] gx modularity and hooks
- [ ] merge add perf
- [ ] go-datastore nosync flatfs
- [ ] addrsplosion part 4

**Comment by harlantwood** [comment]
I'm working on: 

- [ ] ipfs/ci-status#1

**Comment by kyledrake** [comment]
@kyledrake:


- [ ] libp2p site
- [ ] continue on DMCA
- [~] changes to blog post

**Comment by diasdavid** [comment]
## SPRINT CHECK IN

- libp2p
  - [x] Review IPLD
  - [x] make a sync API for js-ipld https://github.com/diasdavid/js-ipld/pull/8
- npm's registry on IPFS (on hold till 0.3.9 rebase on top of 0.4.0)
  - [x] Release a working version of ipfs-blob-store with all of the new shiny things https://github.com/ipfs/ipfs-blob-store/releases/tag/v0.1.0
  - [~] download the entire npm - Almost there [470503/472177]  as the time of this post !
  - [x] Make registry-mirror https://github.com/diasdavid/registry-mirror
- js-ipfs-api
  - [x] fix double recursive flag bug https://github.com/ipfs/js-ipfs-api/pull/110
  - [x] fix, make errors parsable again https://github.com/ipfs/js-ipfs-api/pull/111
- extra
  - [x] Organized and delivered a talk at the 1st IPFS Lisbon. People got very excited and we will probably have a second meetup December with a person from AWMN Mesh community that is doing his PhD in Lisbon + the maintainer of Unimos, the Portuguese large Meshnet, the goal is to have several people setting up OSLR and then use IPFS to distribute data. Will be looking for sponsorship to get cheaper routers.
  - [x] Got invited for NodeConf Barcelona
  - [x] Prepare presentation for NodeConf Barcelona
  - [x] Be a NodeKnockout Judge
  - [x] Flew to Barcelona for the Weekend and attend IPFS Hack Fest Barcelona. Huge hi5 to @masylum and @VictorBjelkholm for organizing it!

**Comment by RichardLitt** [comment]
### @lgierth's sprint update

- go-ipfs
  - [~] RFCR fix gateway links with url rewrites ipfs/go-ipfs#1971 ipfs/infrastructure#116
  - [x] deploy dev0.4.0 to castor and pluto ipfs/infrastructure#115
- infrastructure
  - [x] fix restarter ipfs/infrastructure@6dfef4b
  - [x] update ansible ipfs/infrastructure#114
  - [x] remove ipfs logging config, was thought to be broken, but actually removed! ipfs/go-ipfs#1959
  - [~] open up cors on gateways (allow origin *) ipfs/infrastructure#109
  - [ ] \<hostname>.h.ipfs.io pointing to cjdns addresses
- refs
  - [ ] dmca yyyy/mm/dd structure ipfs/refs#4
  - [x] writeup for kyledrake https://asciinema.org/a/7wgjaa5co0oo1lft5l1tvws43 (the api key in there has since been revoked, my bad)
- cjdns
  - [x] update to latest v17
  - [~] discover peers from cjdns ipfs/go-ipfs#1316
  - [~] invite to fc00 development (aka go-cjdns)
- org stuff
- hope
  - [ ] plan for community pinning / pincoop ipfs/infrastructure#113
  - [ ] plan for writable gateway ipfs/infrastructure#105

**Comment by RichardLitt** [comment]
@VictorBjelkholm's update:

Many things to do over at Typeform, got some time to spend on IPFS though...

### Update

**IPFS in JS**

- [X] Finish writing docs and API docs (Still haven't pushed the branch for this, will do tonight!)
- [ ] Help out to finish the API.md for js-ipfs-api
- [X] working on node-ipfs-api to pass the tests in browser + node env (didn't help very much with this...)

**PinCoop** - https://github.com/victorbjelkholm/pincoop

- [X] Rename to PinCoop
- [ ] having PinCoop ready for public consumption, mainly to scratch my own need for ipfsbin
- [X] Finish prototype with pull-model (did at the IPFS hackday)

**IPFSBin** - https://github.com/victorbjelkholm/ipfsbin

- [X] new ui for ipfsbin to be able to select language amongst other things
- [ ] start using PinCoop for ipfsbin, to make content more available

**nodejs version hosting on IPFS** - https://github.com/victorbjelkholm/nodejs-mirror

- [X] Deploy
- [X] Download all versions
- [X] Publish
- [X] Write blog post about it https://gist.github.com/VictorBjelkholm/af38baab3728414ab478

**Community**

- [~] preparing talk for local meetups here in Barcelona 
- [X] collecting things for awesome-ipfs showcase
- [X] Barcelona hack day with IPFS and Redbooth

**Comment by RichardLitt** [comment]
My update:

I did not have a good on-keyboard week this week, at all. I was AFK for most of it. I feel poorly about that. 

## IFPS HTTP docs
- [ ] Automatically populate a basic template as a one-off seed. Perhaps use ipfs/js-ipfs-api/pull/105.
- [x] Put a Swagger info bloc in ipfs/api.
- [ ] Look at httpie
- [ ] Document how to use curl for your own use to figure out what is missing in the API docs.
- [ ] Add a PR to ipfs.io/docs/api about how the commands there are not all-inclusive.
- [ ] Get `ipfs add` documented.
- [ ] Stub out the rest.
- [ ] Respond to ipfs/api/issues/1.

## READMEs
- [ ] Parse IPFS readme's automatically using a CLI tool for [ipfs-readme-standard](ipfs/ipfs-readme-standard).
- [ ] CR on @victorbjelkholm's ipfs/js-ipfs-api/pull/102
- [ ] Help with ipfs/ci-status/issues/1 and ipfs/infrastructure/issues/25.

## Community
- [x] Add ipfs/ci-status to PD.md. https://github.com/ipfs/ipfs/pull/123
- [x] Sprint management.
- [x] Go through and move ideas from ipfs/pm/issues/43 into ipfs/pm/README.md. https://github.com/ipfs/pm/pull/52
- [x] Edit ipfs/community/blob/master/docs/amending-commits.md. https://github.com/ipfs/community/pull/70
- [ ] Respond to issue about community infrastructure for meetups.
- [ ] Make the documentation around Commit Sign-Offs more friendly
- [ ] Template out issue reporting file from ipfs/go-ipfs/issues/1946.


## Added Done
- [x] Updated project-directory.md PR https://github.com/ipfs/ipfs/pull/116
- [x] Talked a lot about community entry pointes, added various tasks to my TODO. Cf https://github.com/ipfs/community/issues/69

**Comment by RichardLitt** [comment]
## @whyrusleeping sprint update:
- [x] merge transports changes
- [x] rebase 0.4.0
- [x] extract go-libp2p
  - [ ] still working on vendoring everything nicely
- [x] gx modularity and hooks
  - [x] lots of gx fun
- [ ] merge add perf
  - [x] rebased on 0.4.0
- [x] go-datastore nosync flatfs
- [ ] addrsplosion part 4
- [x] dht query profiling
	- [x] kbucket nearest peers fix
	- [ ] dht query perf fixes (waiting to PR)
	- [ ] swarm dial backoff more aggressive (code complete)

**Comment by RichardLitt** [comment]
@Dignifiedquire's update:

- [~] Start mirroring npm-registry
- [~] Investigate and improve performance for `reginabox` and related modules https://github.com/davglass/follow-registry/pull/10
- [x] Refactor js-ipfsd-ctl and add ability to select a specific ipfs binary (needed for station packaging) https://github.com/ipfs/js-ipfsd-ctl/pull/33
- [x] Make ipfs-geoip more resilient https://github.com/ipfs/ipfs-geoip/pull/4
- [x] Use pretty print from ipfs-geoip in webui https://github.com/ipfs/webui/pull/93

**Comment by RichardLitt** [comment]
@kyledrake:

- [x] libp2p site
- [~] continue on DMCA
- [x] changes to blog post

I may be late for the sprint but will be joining the infrastructure hangout.

**Comment by RichardLitt** [comment]
Note: Pasted these above because I accidentally directed everyone to the wrong thread. My bad!

**Comment by jbenet** [comment]
@jbenet
- [x] devcon1 -- this ate all of my time. talks, meeting with users and more.
  - [x] ipfs talk and related conversations
  - [x] 3x interviews about ipfs
  - [x] many conversations. some with:
    - [x] EthDev
    - [x] Consensys
    - [x] Augur
    - [x] Eris
    - [x] other smart contract + ipfs users
    - [x] Floppy (James) at ODI 
    - [x] Ira from Redecentralize
    - [x] Rufus from OKFN
- [x] IPFS London Meetup 

![](https://pbs.twimg.com/media/CTeOCMiWcAARr-m.jpg:large)

**Comment by ion1** [comment]
Update:

* [~] Slight progress with https://github.com/ipfs/go-ipfs/pull/1921, more work remains to be done.

**Comment by masylum** [comment]
# Sprint Check in

* community
  - [x] Co-organized the first Barcelona IPFS meetup. People had a lot of fun and got their hands into it.
* libp2p
  - [~] Started to work on `js-libp2p-repo`, compatible with `abstract-blob-store` which means that we will be able to have ipfs repositories on ipfs repositories on ipfs (you get it).
